splice_machine/src/main/java/com/splicemachine/derby/stream/function/ReturnStatisticsFlatMapFunction.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2012 - 2017 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.splicemachine.derby.stream.function;
+
+import com.splicemachine.derby.impl.sql.execute.operations.LocatedRow;
+import com.splicemachine.derby.utils.StatisticsOperation;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+public class ReturnStatisticsFlatMapFunction
+    extends SpliceFlatMapFunction<StatisticsOperation, Iterator<MergeStatisticsHolder>, LocatedRow> {
+
+    public ReturnStatisticsFlatMapFunction() {
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Iterator<LocatedRow> call(Iterator<MergeStatisticsHolder> holders) throws Exception {
+        return holders.next().toList();
```

Comment:
In the original logic, when we finish merging the column stats from all partitions, we call builder.getValue().terminate(), where builder is of type ColumnStatisticsMerge, but I don't see this function called in the new code. Is it no longer needed? I thought we still need it. Please correct me if I'm wrong.